WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostPagerActivity.java

```diff
@@ -1051,7 +1051,7 @@ public void onPositiveClicked(@NotNull String instanceTag) {
     public void onPostUploaded(OnPostUploaded event) {
         int siteLocalId = AppPrefs.getSelectedSite();
         SiteModel site = mSiteStore.getSiteByLocalId(siteLocalId);
-        if (site != null && event.post != null && event.post.getLocalSiteId() == site.getId()) {
+        if (site != null && event.post != null) {
```

Comment:
This is a good point. I agree that check is an explicit guard condition to avoid to manage posts not related to the current site. I also didn't find an immediate issue removing it so I'm accepting your solution for now. One thing we could maybe do is to check if the event siteId is from a site we are allowed to post to.